internal/controllers/fake/client.go

```diff
@@ -39,6 +45,12 @@ type fakeStatusWriter struct {
 }
 
 func (f fakeStatusWriter) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	if ctxErr := ctx.Err(); ctxErr != nil {
+		// controller-runtime fake ignores context; check it here to allow controllers to test
+		// handling of cancellation related errors
+		return ctxErr
+	}
```

Comment:
FYI There's a non-zero chance this is going to eventually expose races in controller tests 😬